### PR TITLE
fix(plots): honor custom ax in partial_dependence_plot

### LINKED ISSUE #3206: Partial Dependence Plots doesn’t work with custom ax
### Issue Description

Hi! This issue is the same as [this](https://stackoverflow.com/questions/74381596/shap-partial-dependence-subplots-issue) stack Overflow question.

I want to plot different feature pdp in different axes but doing like this it doesn’t work as expected. Only the first feature is plotted but in the last ax. The rest are like overlapping the first.

### Minimal Reproducible Example

```python
import shap
import lightgbm as lgbm
import matplotlib.pyplot as plt
import pandas as pd

# Load the "adult" dataset from shap
X, y = shap.datasets.adult()
features = X.columns

# Split into training and testing sets
df = pd.concat([X, pd.Series(y, name='y_test')], axis=1)

# Your code snippet
X = df.loc[:, features]
X = X.fillna(X.median())
y_test = df.loc[:, 'y_test'].values
model = lgbm.LGBMRegressor()
model.fit(X, y_test)
explainer = shap.TreeExplainer(model)
shap_values = explainer.shap_values(X)

fig, axes = plt.subplots(nrows=3, ncols=3, figsize=(16, 16))
axes_dic = {
    0: axes[0, 0],
    1: axes[0, 1],
    2: axes[0, 2],
    3: axes[1, 0],
    4: axes[1, 1],
    5: axes[1, 2],
    6: axes[2, 0],
    7: axes[2, 1],
    8: axes[2, 2]
}
fig, axes = plt.subplots(nrows=3, ncols=3, figsize=(16, 16))
fig.suptitle('Partial Dependence SHAP plots', fontsize=20)
for i in range(min(len(axes_dic), len(features))):
   shap.partial_dependence_plot(
       features[i], model.predict, X, model_expected_value=True,
       feature_expected_value=True, ice=False, show=False, ax=axes_dic[i]
   )
plt.tight_layout()
plt.show()
```

### Traceback

_No response_

### Expected Behavior

_No response_

### Bug report checklist

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the [latest release](https://github.com/shap/shap/releases) of shap.
- [X] I have confirmed this bug exists on the [master branch](https://github.com/shap/shap/blob/master/CONTRIBUTING.md#installing-from-the-master-branch) of shap.
- [ ] I'd be interested in making a PR to fix this bug

### Installed Versions

0.42.0

### PR DESCRIPTION
**Pull request by A-verse**
**Note for maintainers:** I'm aware of related PRs #4140, #4411, and #4481 which propose similar fixes for this issue. I implemented this independently while learning the codebase as part of my ESoC 2026 application, and I'm sharing it for transparency rather than expecting it to be merged. I'm happy to defer to whichever PR you prefer to merge, or to help test/review the others.

Closes #3206.

## Problem

`shap.partial_dependence_plot(..., ax=my_ax)` ignores the passed `ax`. The `else` branch of the axis setup block calls `plt.gca()` instead of using the caller's axis, so a loop drawing different features into separate subplots ends up stacking everything onto matplotlib's globally-current axis — typically just the last subplot.

## Reproduction

Verified locally on Windows / Python 3.13 / matplotlib 3.10.9 with this snippet:

```python
import matplotlib.pyplot as plt
import numpy as np
import shap

rng = np.random.default_rng(0)
X = rng.standard_normal((30, 3))
def model(x): return x[:, 0] + 0.5 * x[:, 1] - 0.3 * x[:, 2]

fig, axes = plt.subplots(1, 3, figsize=(12, 4))
for i in range(3):
    shap.partial_dependence_plot(i, model, X, ice=False, show=False, ax=axes[i])
```

**Before the fix:**
```
axes[0].has_data(): False
axes[1].has_data(): False
axes[2].has_data(): True
```

**After the fix:**
```
axes[0].has_data(): True
axes[1].has_data(): True
axes[2].has_data(): True
```

## Fix

Two-line change in `shap/plots/_partial_dependence.py`: in the `else` branch of the axis setup, use the passed `ax` directly and recover its figure via `ax.get_figure()` instead of falling back to `plt.gca()`.

## Test

Added `tests/plots/test_partial_dependence.py` with a regression test that creates two subplots, calls `partial_dependence_plot` with `ax=` for each, and asserts both axes have artists. Verified locally that the test fails on master (without this change) and passes with the fix applied.

## Scope

Limited to the 1D path, which is what the issue reports. The 2D (`ind=(a, b)`) surface path also overrides `ax` but fixing it would require the caller to pass a 3D projection axis — different signature contract, kept out of scope.

## Before / After
before

**Before:** Only the last subplot is drawn into.

after

**After:** Each subplot has its own PDP.

### CODE REVIEW

File: tests/plots/test_partial_dependence.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
+"""Tests for shap.partial_dependence_plot."""
+
+import matplotlib
+
+matplotlib.use("Agg")  # headless backend so tests don't pop windows
```
[the remark this answers is not in the document]
Comment (A-verse) [reply tests/plots/test_partial_dependence.py]: Good catch — fixed. Removed the module-level backend selection and relying on mpl_test_cleanup from tests/conftest.py.

File: tests/plots/test_partial_dependence.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    called ``plt.gca()`` in the ``else`` branch and ignored the caller's
+    axis, so multiple plots in a loop all stacked onto the last subplot.
+    """
+    rng = np.random.default_rng(0)
+    X = rng.standard_normal((30, 3))
+
+    def model(x):
+        return x[:, 0] + 0.5 * x[:, 1]
+
+    fig, axes = plt.subplots(1, 2)
```
[the remark this answers is not in the document]
Comment (A-verse) [reply tests/plots/test_partial_dependence.py]: Done — switched to _, axes = plt.subplots(...).
